content/doc/developer/plugin-development/index.adoc

```diff
@@ -0,0 +1,29 @@
+---
+title: Plugin development
+summary: About Jenkins' internal extension functionality, modules, and plugins
+layout: developerchapter
+wip: true
+references:
+- url: https://wiki.jenkins-ci.org/display/JENKINS/Adding+tool+auto-installer
+  title: Adding tool auto-installer
+- url: https://wiki.jenkins-ci.org/display/JENKINS/Choosing+Jenkins+version+to+build+against
+  title: Choosing Jenkins version to build against
+- url: https://wiki.jenkins-ci.org/display/JENKINS/Dependencies+among+plugins
+  title: Dependencies among plugins
+- url: https://wiki.jenkins-ci.org/display/JENKINS/Tips+for+optional+dependencies
+  title: Tips for optional dependencies
+- url: https://wiki.jenkins-ci.org/display/JENKINS/Making+your+plugin+work+with+blue+ocean
+  title: Making your plugin work with Blue Ocean
```

Comment:
Should be straightforward to pull out the larger sections of plugin development into top-level chapters once we're far enough along.